spec/rspec/core/configuration_spec.rb

```diff
@@ -1229,7 +1229,167 @@ def metadata_hash(*args)
       end
     end
 
+    describe "#color_mode" do
+      context ":automatic" do
+        before do
+          config.color_mode = :automatic
+        end
+
+        context "with output.tty?" do
+          it "sets color_enabled?" do
+            output = StringIO.new
+            config.output_stream = output
+
+            allow(config.output_stream).to receive_messages(:tty? => true)
+
+            expect(config.color_enabled?).to be true
+            expect(config.color_enabled?(output)).to be true
```

Comment:
In all of the examples for `color_mode` you have two paired expectations like this:

``` ruby
expect(config.color_enabled?).to be value
expect(config.color_enabled?(output)).to be value
```

...where `value` is always the same.  I'm realizing that as written, the implementation could completely ignore the argument, because the passed arg has the same `tty?` status as the configured `output_stream` (and in fact, it's the very same IO object!).

Since it's important that `color_enabled?` does in fact check `tty?` on the passed arg, and does not always check `output_stream`, it would be nice if the specs would cover this case.  Can you restructure the existing specs to cover this or add a new one to cover it?